evm/chains/base.py

```diff
@@ -686,15 +687,26 @@ def validate_uncles(self, block: BaseBlock) -> None:
             uncle_vm_class = self.get_vm_class_for_block_number(uncle.block_number)
             uncle_vm_class.validate_uncle(block, uncle, uncle_parent)
 
-    def validate_chain(self, chain: Tuple[BlockHeader, ...]) -> None:
+    def validate_chain(self, chain: Tuple[BlockHeader, ...], seal_check_frequency: int = 1) -> None:
         parent = self.chaindb.get_block_header_by_hash(chain[0].parent_hash)
-        for header in chain:
+        all_indices = list(range(len(chain)))
+        if seal_check_frequency == 1:
+            headers_to_check_seal = all_indices
+        else:
+            headers_to_check_seal = random.sample(
+                all_indices, len(all_indices) // seal_check_frequency)
```

Comment:
This looks like it's written as if the seal_check constant is a period. Otherwise, `192 // (192/4)` would only check 4 headers total in a chain of 192.

Reply:
Yes, that's what I intended. In geth they check every 100-th header, and we're doing every 48th